src/app/workflow/workflow-file-editor/workflow-file-editor.component.ts

```diff
@@ -64,10 +64,14 @@ export class WorkflowFileEditorComponent extends FileEditing {
     if (!this.versions || this.versions.length === 0) {
       return true;
     }
-    const mostRecentId = this.versions.reduce((max, n) => Math.max(max, n.id), this.versions[0].id);
+    const mostRecentId = this.getIdOfNewestVersion(this.versions);
     return this._selectedVersion.id === mostRecentId;
   }
 
+  getIdOfNewestVersion(versions: Array<WorkflowVersion>): number {
+    return versions.reduce((max, n) => Math.max(max, n.id), this.versions[0].id);
```

Comment:
This code assumes at least one version. I guess you're safe, since the previous invocation checks for it, and the new invocation will be invoked only after a successful save.